static/js/message_helper.js

```diff
@@ -0,0 +1,78 @@
+import * as alert_words from "./alert_words";
+import * as message_store from "./message_store";
+import * as message_user_ids from "./message_user_ids";
+import * as people from "./people";
+import * as recent_senders from "./recent_senders";
+import * as stream_topic_history from "./stream_topic_history";
+import * as util from "./util";
+
+export function process_new_message(message) {
```

Comment:
This extraction is great, but I'm not sure `message_helper` is the right name.  What do you think about `message_metadata.process_new_message`?  

I could imagine moving a future set of functions related to message editing here as well.